src/pal/src/misc/time.cpp

```diff
@@ -270,6 +275,29 @@ GetTickCount64()
         }
         retval = (ts.tv_sec * tccSecondsToMillieSeconds)+(ts.tv_nsec / tccMillieSecondsToNanoSeconds);
     }
+#elif HAVE_MACH_CLOCK_MONOTONIC
+    {
+        kern_return_t machRet;
+        clock_serv_t clock;
+        mach_timespec_t mts;
+        host_t host = mach_host_self();
+
+        if((machRet = host_get_clock_service(host, SYSTEM_CLOCK, &clock)) != KERN_SUCCESS)
```

Comment:
The microbenchmark is not right. GetTimeofDay() method should not be called for every iteration because of it is about as expensive as Environment.TickCount that you are trying to measure. It should be called only once before and after the loop:

```
        public static void TickCountPerf()
        {            
            int iterations = 10000000;
            long startTime = GetTimeofDay();
            for (int i = 0; i < iterations; i++)
            {
                int tick = Environment.TickCount;
            }
            long totalTime = GetTimeofDay() - startTime;
            ...
    }
```

Once you fix the microbenchmark, I expect that you will get numbers ~0.03 microseconds without your change, and ~3 microseconds with your change. Making Environment.TickCount 100x slower is certainly not within acceptable level of tolerance.

For reference, Environment.TickCount takes less than 0.01 microseconds on Windows.